hoover_script.py

```diff
@@ -30,9 +30,167 @@
     raise RuntimeError("HOOVER_HOME environment variable is not set")
 home = Path(_home)
 
-def question(label, default):
-    rv = input("{} [{}]: ".format(label, default))
-    return rv.strip() or default
+interactive_mode = False
+
+class Param:
+    param_list = []
+
+    def __init__(self, name, default, environ=None, question_label=None, optional=False):
```

Comment:
`optional=False` is like double negation, I think `required=True` is easier to understand.